components/director/internal/domain/formation/resolver.go

```diff
@@ -149,3 +169,32 @@ func (r *Resolver) UnassignFormation(ctx context.Context, objectID string, objec
 
 	return r.conv.ToGraphQL(newFormation), nil
 }
+
+func (r *Resolver) assignFormationForTenant(ctx context.Context, tx *persistence.PersistenceTx, tenantFromContext string, objectID string, formationInput graphql.FormationInput) (*model.Formation, error) {
```

Comment:
Can you move this logic in the service instead of the resolver?

We are planning to use the formation service in other flows as well and this logic fits more like a business logic instead of graphQL API logic.

Also, this will remove the if above and the differentiation between the formation object type will be only in the service.